### ISSUE TITLE
Hibernate: merge detached entities with inheritance ignore null attributes on child classes

### ISSUE THREAD

**Issue by antoinechamot** [quarkusio/quarkus#39654]
### Describe the bug

Using Hibernate entity inheritance ( Quarkus 3 ).

I have created a repository with a test class to reproduce the bug: https://github.com/antoinechamot/quarkus-bugfind. Launch tests (one must fail )

My entities are 

```java
@SuperBuilder
@NoArgsConstructor
@Setter
@Getter
@Inheritance(strategy = InheritanceType.JOINED)
@Entity(name = "shape")
public abstract class Shape {

    @Id
    @Builder.Default
    private UUID id = UUID.randomUUID();

    @Column(nullable = false, length = 100)
    private String name;

    @Column(length = 100)
    private String surname;
}
```

```java
@Entity(name = "circle")
@SuperBuilder
@NoArgsConstructor
@Getter
@Setter
public class Circle extends Shape{

    private String color;
}
```

I have a simple repository with one method
```java
@Override
@Transactional
public Circle update(Circle circle) {
    return getEntityManager().merge(circle);
}
```

For test i populate the database with some data:
```sql
INSERT INTO shape (id,name,surname) VALUES ('847dae35-7ae0-4041-8257-e23fb7c5b033','My circle','My surname');
INSERT INTO circle (id,color) VALUES ('847dae35-7ae0-4041-8257-e23fb7c5b033','red');
...
```

And then i run a test to update the circle color to null

```java
    /**
     * Update Circle color to null : KO
     */
    @Test
    void testUpdateCircleColorNullKO(){
        Circle myCircle = Circle.builder()
                .id(UUID.fromString("847dae35-7ae0-4041-8257-e23fb7c5b033"))
                .name("My circle")
                .surname("My surname")
                .color(null)
                .build();

        repository.update(myCircle);

        Circle updatedCircle = repository.findById(UUID.fromString("847dae35-7ae0-4041-8257-e23fb7c5b033"));
        assertThat(updatedCircle.getColor()).isNull();
    }

```

 

### Expected behavior

- On merge, I expect hibernate to launch an update request to set the value in the color column of the circle table to null.
- If I put null on surname for example, an update request is launched by hibernate.
- It seems that the problem only occurs on attributes of the child class Circle and not on the parent class Shape 

### Actual behavior

- On merge, null attributes of the child class Circle are ignored. Not update request is done on the circle table.
- If I update the color to something not null like 'blue', it is correctly updated

### How to Reproduce?

Repository : https://github.com/antoinechamot/quarkus-bugfind

**Comment by yrodiere** [comment]
Hello,

Is this different from https://github.com/quarkusio/quarkus/issues/38722 ? Because that one is a bug in Hibernate ORM, not in Quarkus, and it's already being fixed upstream.

If it's not the same bug, you might want to check these already-reported issues that mention "merge": https://github.com/quarkusio/quarkus/issues?q=is%3Aissue+is%3Aopen+merge+label%3Aarea%2Fhibernate-orm

**Comment by yrodiere** [comment]
I am referring to https://hibernate.atlassian.net/browse/HHH-17761 and the corresponding PR https://github.com/hibernate/hibernate-orm/pull/7912 , yes.

**Comment by yrodiere** [comment]
@antoinechamot I'll go ahead and close this as duplicate of #38722, which has been fixed on main (Quarkus 3.11). Feel free to comment here if it turns out this was a different problem.
Thanks for your feedback.
